pytorch_lightning/trainer/training_loop.py

```diff
@@ -355,12 +356,14 @@ def train(self):
                 stop = should_stop and met_min_epochs
                 if stop:
                     self.main_progress_bar.close()
-                    model.on_train_end()
+                    with self.profiler.profile('on_train_end'):
+                        model.on_train_end()
                     return
 
         self.main_progress_bar.close()
 
-        model.on_train_end()
+        with self.profiler.profile('on_train_end'):
```

Comment:
just thinking is some form of decorator would be cleaner...

Reply:
we discussed it s lot. we’ll keep it without decorators for now